packages/flame/lib/src/components/base_component.dart

```diff
@@ -4,30 +4,16 @@ import 'package:meta/meta.dart';
 
 import '../../game.dart';
 import '../../input.dart';
-import '../effects/effects.dart';
-import '../effects/effects_handler.dart';
 import '../extensions/vector2.dart';
 import '../text.dart';
 import 'component.dart';
-import 'component_set.dart';
-import 'mixins/has_game_ref.dart';
 
 /// This can be extended to represent a basic Component for your game.
 ///
-/// The difference between this and [Component] is that the [BaseComponent] can
-/// have children, handle effects and can be used to see whether a position on
-/// the screen is on your component, which is useful for handling gestures.
-abstract class BaseComponent extends Component {
-  final EffectsHandler _effectsHandler = EffectsHandler();
-
-  late final ComponentSet children = createComponentSet();
-
-  /// If the component has a parent it will be set here
-  BaseComponent? _parent;
-
-  @override
-  BaseComponent? get parent => _parent;
-
+/// The difference between this and [Component] is that the [BaseComponent]
+/// implements the [update] and [render] methods, meanwhile you have to do that
+/// yourself in [Component].
+class BaseComponent extends Component {
```

Comment:
if we are moving the children up and this is just a shin for propagating update and render, maybe we can just get rid of it?

Reply:
I thought about that too, the only extra thing that it has now is the debug stuff and that might be worth having on the `Component` anyways? What do you think @flame-engine/flame-colab?

Reply:
that should be extracted to a separate file imo, and then it can be "hooked up" with a couple lines at any level we desire. I think it's worth it